app/services/nutritionix/api_client.rb

```diff
@@ -0,0 +1,105 @@
+require 'httparty'
+
+module Nutritionix
+
+# ---------------------------can't get calories
+# class ApiClient
+#   include HTTParty
+#   format :json
+#   base_uri "https://api.nutritionix.com"
+#
+#   def initialize()
+#   end
+#
+#   def get(product)
+#     api_key = Rails.application.credentials.dig(:nutritionix, :api_key)
+#     app_id = Rails.application.credentials.dig(:nutritionix, :app_id)
+#     params = {
+#               headers: { "Content-Type":"application/json" },
+#               body: [{ appId: app_id,
+#                       appKey: api_key,
+#                       fields: ["item_name", "brand_name", "nf_calories", "nf_serving_weight_grams", "nf_serving_size_qty"]
+#                     }].to_json
+#               }
+#     response = self.class.get("/v1_1/search/#{product.parameterize.underscore}?", params).parsed_response
+#     @product = response["hits"][0]
+#   end
+#
+#   def calories
+#     @product["fields"]["nf_calories"]
+#   end
+#
+#   def grams
+#     @product["fields"]["nf_serving_weight_grams"]
+#   end
+#
+#   def quantity
+#     @product["fields"]["nf_serving_size_qty"]
+#   end
+#
+# end
+
+# ---------------------------works, but not sure if this is correct, pretty way
+class ApiClient
+  include HTTParty
+  format :json
+  base_uri "https://api.nutritionix.com"
+
+  def initialize(product)
+    api_key = Rails.application.credentials.dig(:nutritionix, :api_key)
+    app_id = Rails.application.credentials.dig(:nutritionix, :app_id)
+    response = HTTParty.get("https://api.nutritionix.com/v1_1/search/#{product.parameterize.underscore}?results=0:2&fields=item_name,brand_name,item_id,nf_calories,nf_serving_weight_grams&appId=#{app_id}&appKey=#{api_key}")
```

Comment:
I would also keep in the constructor only initial settings like api_key and app_id and made separate `search` method, where actual get request could be done. also please don't pass manually all parameters in url.
something like this should work:
```
HTTParty.get("url goes here", query: { result: 0, fields: [:item_name, :brand_name])
```